src/full/Agda/TypeChecking/Monad/Builtin.hs-boot

```diff
@@ -0,0 +1,28 @@
+{-# LANGUAGE CPP #-}
+
+module Agda.TypeChecking.Monad.Builtin where
+
+import Control.Monad.Reader
+import Control.Monad.State
+
+#if __GLASGOW_HASKELL__ >= 800
+import qualified Control.Monad.Fail as Fail
+#endif
+import Control.Monad.IO.Class (MonadIO)
+
+import Agda.TypeChecking.Monad.Base (TCMT, Builtin, PrimFun)
+
+class ( Functor m
+      , Applicative m
+#if __GLASGOW_HASKELL__ == 710
+      , Monad m
+#else
+      , Fail.MonadFail m
+#endif
```

Comment:
These preprocessor conditions can be removed now that 7.x support is gone, and that means `LANGUAGE CPP` above can be removed. Since that's the case in a lot of these files, I'll stop commenting on these now 👍

Reply:
I think I removed all of them now